### PR TITLE
[SMALLFIX] Add script for publishing release tarballs

### PR DESCRIPTION
**Pull request by aaudiber**
This PR is ready for review, we shouldn't merge it right away though

### CODE REVIEW

File: dev/scripts/generate-tarball.sh [hunk 1]
```diff
@@ -89,8 +95,13 @@ function build {
...
 function create_tarball {
   cd "${REPO_COPY}"
+  local hadoop_profile=$1
   local version="$(bin/alluxio version)"
-  local prefix="alluxio-${version}"
+  if [[ "${hadoop_profile}" == "default" ]]; then
+    local prefix="alluxio-${version}-bin"
+  else
+    local prefix="alluxio-${version}-${hadoop_profile}-bin"
+  fi
```
Comment (calvinjia) [inline dev/scripts/generate-tarball.sh]: Could you also add a line to generate a md5sum?

**Comment by aaudiber** [comment]
We've been avoiding bringing in Go as a dependency in this repo. Users are much more likely to have Bash

**Comment by jsimsa** [comment]
do you expect users ever needing to invoke these scripts?

**Comment by aaudiber** [comment]
@jsimsa Probably not the publish script, but definitely the generate-tarball script. The Docker build script depends on it, and users may want to use the build script to create Docker images.

File: dev/scripts/generate-tarball.sh (hunk 1 above)
Comment (aaudiber) [inline dev/scripts/generate-tarball.sh]: done

**Comment by aaudiber** [comment]
@calvinjia Comments addressed, PTAL

**Pull request merged**